### PR TITLE
Add a disputes task to the "Things to do next" list on WC > Home

### LINKED ISSUE #4341: Add dispute task to the "Things to do next" on the WC → Home screen for disputes which need a response
### Description

To draw store managers' attention to the disputes which need of a response, we're going to add a new task item to the **`Things to do next`** task list. Clicking this task item will direct the user to the disputes screen. This was proposed and agreed upon in this P2 proposal (pdjTHR-19j-p2) and the subsequent accepted answer (pdjTHR-19j-p2#comment-1404). 

### Acceptance criteria

* Add the Disputed payment task item to the **WooCommerce `Things to do next`** task list.
* Clicking on this task, navigates the user to the Disputes screen (filtered by disputes needing a response). 

### Designs

image

### Dev notes

* [WooCommerce add task example](https://github.com/woocommerce/woocommerce/blob/71456382c47eadaa4bb98501bd3c7aafbfde1a52/plugins/woocommerce-admin/docs/examples/extensions/add-task/woocommerce-admin-add-task-example.php)
* There's a WIP branch which can be found here: https://github.com/Automattic/woocommerce-payments/compare/develop...add/4341-add-disputes-task-to-thing-to-do-next

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #4341

#### Changes proposed in this Pull Request

This PR adds a task item to the **WC → Home** matching the one on **Payments → Overview**. 

Screen Shot 2022-07-07 at 2 19 52 pm

#### Testing instructions

1. Purchase a product using a disputed payment method. eg `4000000000000259`
2. Go to **WooCommerce → Home** on your admin dashboard. 
3. Note the task item referencing the disputed payment. 
4. Clicking this task should navigate to the Disputes list table with the **"Needs response"** filter applied. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by Jinksi** [comment]
I've noticed that the task will show a stale dispute count after accepting a dispute and navigating back to WC → Home. 

This should be fixed alongside https://github.com/Automattic/woocommerce-payments/issues/4373.

**Review by Jinksi — APPROVED** [review]
This looks good to me @james-allan,  🙌  works as expected and matches the WCPay Overview disputes task (updated in #4388).

There is a small difference with the subtext: this WC Home task has `30 minutes` and the WCPay task has `View and respond`, however, this is likely to be updated again soon to match the proposed design.

One thought I had while testing: is it feasible/suitable to de-duplicate the text between the 2 disparate dispute tasks (WC Home vs WCPay Overview) or link them to each other somehow? My concern is that it won't be obvious to future devs that the 2 tasks are defined in completely different parts of the codebase, even though they look identical.

**Steps tested:**
- Installed on a fresh JN site. The task isn't visible by default.
- Created a disputed transaction, disputes task shows the correct count and links to 
- Accepted the dispute. Disputes task count is stale until page refresh - see https://github.com/Automattic/woocommerce-payments/pull/4418#issuecomment-1177147236. On refresh, the task is no longer visible, as expected.
- Tested with WC 6.6.1: works as expected.
- Tested with WC 6.2.0: works as expected.
- Tested with WC 5.8.0: task doesn't display but no errors or warnings (expected not to appear as WC TaskList API was introduced in a later version)

**Comment by james-allan** [comment]
> There is a small difference with the subtext: this WC Home task has 30 minutes and the WCPay task has View and respond, however, this is likely to be updated again soon to match the proposed design.

Thanks for highlighting this. I planned to look at what options we have for this time estimation but forgot before submitting this PR. I just plucked 30 minutes out as the original 2 minutes didn't feel right. 

I've remove this time estimation in [5a9e1e7](https://github.com/Automattic/woocommerce-payments/pull/4418/commits/5a9e1e78c398f863b583e8df2f5e08a06c6692b6) so this PR simply just copies the existing task and and a clean base for the upcoming design discussion.

The task renders fine without a time estimation too: 

Screen Shot 2022-07-08 at 11 23 22 am

> One thought I had while testing: is it feasible/suitable to de-duplicate the text between the 2 disparate dispute tasks (WC Home vs WCPay Overview) or link them to each other somehow? My concern is that it won't be obvious to future devs that the 2 tasks are defined in completely different parts of the codebase, even though they look identical.

We'd need to add a function that returns the strings and then to inject it in the JS you'd need to add an api to fetch it or inject them when the script is localized with variables. So it would be possible, I'm not sure how much work that would be. For example if we went with this approach we might as well change the one on the Payments > Overview screen to be injected completely via localised variables rather than have its content generated in JS. This is probably my php bias talking though as I don't know what options we'd have to do this via JS.

File: changelog/add-wc-home-dispute-task [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: add
+
+Add a task to the WooCommerce > Home screen notifying merchants of disputed payments which need a response.
```
[the remark this answers is not in the document]
Comment (james-allan) [reply changelog/add-wc-home-dispute-task:4]: Thanks Eric and Grammarly!

**Comment by Jinksi** [comment]
> So it would be possible, I'm not sure how much work that would be.

@james-allan What do you think about instead adding a code comment to direct attention to the fact that these tasks are separate?

Something like: `Note: this task is separate to the Payments → Overview disputes task, which is defined in client/overview/task-list/tasks.js.`

**Comment by james-allan** [comment]
> What do you think about instead adding a code comment to direct attention to the fact that these tasks are separate?

Sounds good. Added in [6755683](https://github.com/Automattic/woocommerce-payments/pull/4418/commits/67556832ffb80ccd3079a4049cd2e7c981379eae). :)

**Pull request merged**